From felipe.balbi@nokia.com  Fri Apr 23 14:05:27 2010
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Thu, 25 Mar 2010 13:14:30 +0200
Subject: MUSB: DaVinci: fix musb_platform_init() error cleanup path
To: Greg KH <greg@kroah.com>
Message-ID: <1269515673-27980-9-git-send-email-felipe.balbi@nokia.com>


From: Sergei Shtylyov <sshtylyov@ru.mvista.com>

This function forgets to call clk_disable() iff reading the USB module version
register returns 0.

Signed-off-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/davinci.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/musb/davinci.c
+++ b/drivers/usb/musb/davinci.c
@@ -444,6 +444,8 @@ int __init musb_platform_init(struct mus
 	return 0;
 
 fail:
+	clk_disable(musb->clock);
+
 	usb_nop_xceiv_unregister();
 	return -ENODEV;
 }
